lib/src/pgn.dart

```diff
@@ -0,0 +1,1006 @@
+  final List<Game<PgnNodeData>> games = [];
+  PgnParser((Game<PgnNodeData> game, [Exception? err]) => games.add(game),
+          initHeaders)
+      .parse(pgn);
+  return games;
+}
+
+/// Parse a string for a variant if exist or return null
+Variant? parseVariant(String variant) {
+  switch (variant.toLowerCase()) {
+    case 'chess':
+    case 'chess960':
+    case 'chess 960':
+    case 'standard':
+    case 'from position':
+    case 'classical':
+    case 'normal':
+    case 'fischerandom': // Cute Chess
+    case 'fischerrandom':
+    case 'fischer random':
+    case 'wild/0':
+    case 'wild/1':
+    case 'wild/2':
+    case 'wild/3':
+    case 'wild/4':
+    case 'wild/5':
+    case 'wild/6':
+    case 'wild/7':
+    case 'wild/8':
+    case 'wild/8a':
+      return Variant.chess;
+    case 'crazyhouse':
+    case 'crazy house':
+    case 'house':
+    case 'zh':
+      return Variant.crazyhouse;
+    case 'king of the hill':
+    case 'koth':
+    case 'kingofthehill':
+      return Variant.kingofthehill;
+    case 'three-check':
+    case 'three check':
+    case 'threecheck':
+    case 'three check chess':
+    case '3-check':
+    case '3 check':
+    case '3check':
+      return Variant.threecheck;
+    case 'antichess':
+    case 'anti chess':
+    case 'anti':
+      return Variant.antichess;
+    case 'atomic':
+    case 'atom':
+    case 'atomic chess':
+      return Variant.atomic;
+    case 'horde':
+    case 'horde chess':
+      return Variant.horde;
+    case 'racing kings':
+    case 'racingkings':
+    case 'racing':
+    case 'race':
+      return Variant.racingKings;
+    default:
+      return null;
+  }
+}
+
+/// Create a [Position] from setup and variants
+Position setupPosition(Variant rules, Setup setup,
+    {bool? ignoreImpossibleCheck}) {
+// TODO:missing horde, racingkings. Returns Chess for those variants
+  switch (rules) {
+    case Variant.chess:
+      return Chess.fromSetup(setup,
+          ignoreImpossibleCheck: ignoreImpossibleCheck);
+    case Variant.antichess:
+      return Antichess.fromSetup(setup,
+          ignoreImpossibleCheck: ignoreImpossibleCheck);
+    case Variant.atomic:
+      return Atomic.fromSetup(setup,
+          ignoreImpossibleCheck: ignoreImpossibleCheck);
+    case Variant.kingofthehill:
+      return KingOfTheHill.fromSetup(setup,
+          ignoreImpossibleCheck: ignoreImpossibleCheck);
+    case Variant.crazyhouse:
+      return Crazyhouse.fromSetup(setup,
+          ignoreImpossibleCheck: ignoreImpossibleCheck);
+    case Variant.threecheck:
+      return ThreeCheck.fromSetup(setup,
+          ignoreImpossibleCheck: ignoreImpossibleCheck);
+    default:
+      return Chess.fromSetup(setup,
+          ignoreImpossibleCheck: ignoreImpossibleCheck);
+  }
+}
+
+/// Create a [Position] for a Variant from the headers
+///
+/// Headers must include a 'Variant' and 'Fen' key
+Position startingPosition(Headers headers, {bool? ignoreImpossibleCheck}) {
```

Comment:
same, not sure it's needed and it doesn't belong here

Reply:
Moved to `Position` class